### PR TITLE
New: Add `summary` fomatter

### PR DESCRIPTION
**Pull request by molant**
While I was working on www.tmz.com I got upset with the amount of output from the other formatters so I created a new one that just tells you how good/bad you are doing with all the rules.
Tests are missing but will add once the tmz PR is finished.

### CODE REVIEW

File: src/lib/formatters/summary/summary.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+/**
+ * @fileoverview The summary formatter, it outputs the aggregation of all the rule results in a table format.
+ *
```
Comment (alrra) [inline src/lib/formatters/summary/summary.ts]: Remove empty line.

File: src/lib/formatters/summary/summary.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                return -1;
+            }
+
+            if (problemsA.length > problemsB.length) {
+                return 1;
+            }
+
+            return ruleA.localeCompare(ruleB);
+        });
+
```
Comment (alrra) [inline src/lib/formatters/summary/summary.ts]: Remove empty line.

File: src/lib/formatters/summary/summary.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            totalErrors += errors;
+            totalWarnings += warnings;
+        });
+
+        logger.log(table(tableData));
+
+        const color: chalk.ChalkChain = totalErrors > 0 ? chalk.red : chalk.yellow;
+
+        logger.log(color.bold(`\u2716 Found a total of ${totalErrors} ${pluralize('error', totalErrors)} and ${totalWarnings} ${pluralize('warning', totalWarnings)}`));
```
Comment (alrra) [inline src/lib/formatters/summary/summary.ts]: > \u2716

Maybe use [`log-symbols`](https://github.com/sindresorhus/log-symbols#readme)?

**Comment by alrra** [comment]
Maybe post a screenshot with how the output looks. Thanks!

**Comment by molant** [comment]
image

The missing character is because my command line doesn't support emojis

**Comment by molant** [comment]
@alrra all changes and tests are in

**Pull request closed without merging**